### DISCUSSION TITLE
strange order of db migration job launch

### DISCUSSION THREAD

**Discussion by jaegwonseo** [apache/airflow#28551] [General]
### Official Helm Chart version

1.7.0 (latest released)

### Apache Airflow version

2.4

### Kubernetes Version

1.2.3

### Helm Chart configuration

migrateDatabaseJob:
  enabled: true

### What happened

db migration job starts after all airflow pods launch 

### What you think should happen instead

db migration job starts before airflow pods launch  

### Are you willing to submit PR?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/apache/airflow/blob/main/CODE_OF_CONDUCT.md)

**Comment by jaegwonseo** [comment]
helm hook at migrate-database-job.yaml should be **pre-xxx**
```yaml
  {{- if .Values.migrateDatabaseJob.useHelmHooks }}
    {{- $_ := set $annotations "helm.sh/hook" "pre-install,pre-upgrade" }}
    {{- $_ := set $annotations "helm.sh/hook-weight" "1" }}
    {{- $_ := set $annotations "helm.sh/hook-delete-policy" "before-hook-creation,hook-succeeded" }}
  {{- end }}
```

**Reply by potiuk** [reply]
> helm hook at migrate-database-job.yaml should be **pre-xxx**
> 
> ```yaml
>   {{- if .Values.migrateDatabaseJob.useHelmHooks }}
>     {{- $_ := set $annotations "helm.sh/hook" "pre-install,pre-upgrade" }}
>     {{- $_ := set $annotations "helm.sh/hook-weight" "1" }}
>     {{- $_ := set $annotations "helm.sh/hook-delete-policy" "before-hook-creation,hook-succeeded" }}
>   {{- end }}
> ```

You cannot do that. Before you need to upgrade the image first in order to do the migration to the newer version. So `pre-` is not possible.

**Comment by potiuk** [comment]
How? When ? Logs? Where do you have impression that it happen ? 

Maybe you missed the fact that each airflow POD has "wait-for-migrations" initi container - so they start at THE SAME TIME as migration job, and the init containers waits until migration is complete before **actually** running the pods.
